trace-dispatcher/doc/trace-dispatcher.md

```diff
@@ -108,689 +89,654 @@ data TraceAddBlockEvent blk =
+### MetaTrace Typeclass
 
-The following __trace combinators__ affect __annotated severity__ of a trace:
+The `MetaTrace` typeclass plays a pivotal role in providing meta-information for trace messages. It includes the following methods:
 
-```haskell
-withSeverity :: Monad m => (a -> Severity) -> Trace m a -> Trace m a
-setSeverity  :: Monad m => Severity        -> Trace m a -> Trace m a
+- __namespaceFor__: Offers a distinct (inner) namespace for each trace message.
 
--- Unconditional annotation:
-tracer   = setSeverity Notice trAddBlock
+- __severityFor__: Provides severity for a given namespace. As some severities depend not only on the message type but also on the individual message, the actual message may be passed as well.
 
--- Conditional annotation:
-tracer'' = withSeverity (\case
-                           IgnoreBlockOlderThanK{}          -> Warning
-                           IgnoreBlockAlreadyInVolatileDB{} -> Notice)
-                        trAddBlock
-```
+- __privacyFor__: Determines whether a message is `Private` or `Public`. Private messages are not sent to `cardano-tracer` and are only displayed on the stdout trace. If no implementation is given, `Public` is chosen.
+
+- __detailsFor__: Specifies the level of details for printing messages. Options include `DMinimal`, `DNormal`, `DDetailed`, and `DMaximum`. If no implementation is given, `DNormal` is chosen.
 
-If the combinators are applied multiple times to a single trace, only the outermost application affects it -- the rest of them is ignored.
+- __documentFor__: Allows the addition of optional documentation for messages as text.
+
+- __metricsDocFor__: Enables the addition of documentation for metrics carried by the respective message. If no implementation is given, the default is no metrics.
+
+- __allNamespaces__: Must return an array with all namespaces of this trace type.
 
 ```haskell
-traceWith (setSeverity Warning trAddBlock) (IgnoreBlockOlderThanK b)
-```
+class MetaTrace a where
+  namespaceFor  :: a -> Namespace a
 
-In addition, the __severity context__ of a particular trace can be further overridden by configuration, at the __tracer namespace__ granularity -- which allows to put them above or below the __global severity cutoff__, effectively either enabling or disabling them.
+  severityFor   :: Namespace a -> Maybe a -> Maybe SeverityS
 
-`Info` is the default __severity__, in the absence of trace context or configured severity overrides.
+  privacyFor    :: Namespace a -> Maybe a -> Maybe Privacy
+  privacyFor _ _ =  Just Public
 
-NOTE: as en extension to the filtering severity type (`SeverityF`), a `Silence` constructor is defined, which encodes unconditional silencing of a particular trace -- and therefore serves as a semantic expression of the disabling tracers functionality of the old framework.
+  detailsFor    :: Namespace a -> Maybe a -> Maybe DetailLevel
+  detailsFor _ _ =  Just DNormal
 
-### Privacy
+  documentFor   :: Namespace a -> Maybe Text
```

Comment:
No default implementation like `const Nothing`? Do I have to provide that explicitly as a dev?